Sources/AsyncKit/ConnectionPool/EventLoopConnectionPool.swift

```diff
@@ -36,7 +41,7 @@ public final class EventLoopConnectionPool<Source> where Source: ConnectionPoolS
     
     /// All requests for a connection that were unable to be fulfilled
     /// due to max connection limit having been reached.
-    private var waiters: CircularBuffer<(Logger, EventLoopPromise<Source.Connection>)>
+    private var waiters: CircularBuffer<(Logger, EventLoopPromise<Source.Connection>, UUID)>
```

Comment:
ELPromise.value is a reference type, so you can use `===` to compare instead of storing a UUID.